packages/contracts-bedrock/src/universal/interfaces/IOptimismMintableERC20.sol

```diff
@@ -1,31 +1,80 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
 
 /// @title IOptimismMintableERC20
 /// @notice This interface is available on the OptimismMintableERC20 contract.
 ///         We declare it as a separate interface so that it can be used in
 ///         custom implementations of OptimismMintableERC20.
-interface IOptimismMintableERC20 is IERC165 {
+interface IOptimismMintableERC20 is ISemver {
+    event Mint(address indexed account, uint256 amount);
+    event Burn(address indexed account, uint256 amount);
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+
+    function supportsInterface(bytes4 _interfaceId) external pure returns (bool);
+
+    function DOMAIN_SEPARATOR() external view returns (bytes32);
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    )
+        external;
+
+    function symbol() external view returns (string memory);
+
+    function name() external view returns (string memory);
+
+    function decimals() external view returns (uint8);
+
+    function nonces(address owner) external view returns (uint256);
+
+    function transfer(address to, uint256 amount) external returns (bool);
+
+    function allowance(address _owner, address _spender) external view returns (uint256);
+
+    function approve(address spender, uint256 amount) external returns (bool);
+
+    function transferFrom(address from, address to, uint256 amount) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function decreaseAllowance(address spender, uint256 subtractedValue) external returns (bool);
+
+    function increaseAllowance(address spender, uint256 addedValue) external returns (bool);
+
+    function totalSupply() external view returns (uint256);
+
+    function PERMIT2() external pure returns (address);
+
+    function REMOTE_TOKEN() external view returns (address);
+
+    function BRIDGE() external view returns (address);
+
     function remoteToken() external view returns (address);
 
-    function bridge() external returns (address);
+    function bridge() external view returns (address);
 
     function mint(address _to, uint256 _amount) external;
 
     function burn(address _from, uint256 _amount) external;
-}
 
-/// @custom:legacy
-/// @title ILegacyMintableERC20
-/// @notice This interface was available on the legacy L2StandardERC20 contract.
-///         It remains available on the OptimismMintableERC20 contract for
-///         backwards compatibility.
-interface ILegacyMintableERC20 is IERC165 {
```

Comment:
Same as above, need to revert to original broken interface but let's still put this in a separate file